### ISSUE TITLE
Question: How to change primary user id on the key?

### ISSUE THREAD

**Issue by IvanPizhenko** [vanitasvitae/pgpainless#200]
I want to implement following scenarios with PGPainless:

1. There is key with one or more User Ids, say A, B, C. I want to add new User Id D so that it becomes primary, i.e. D, A, B, C.
2. There is key with two or more user Ids, say A, B, C. I want to make B or C primary User Id, i.e. B, A, C or C, A, B.

How do I do it with PGPainless?

**Comment by vanitasvitae** [comment]
This is currently not possible directly with PGPainless. #195 would probably make implementing this more easy.

**Comment by vanitasvitae** [comment]
Could you take a look at https://github.com/pgpainless/pgpainless/discussions/202?

**Comment by vanitasvitae** [comment]
With the upcoming API this would look something along the lines of
```
PGPSecretKeyRing secretKeys = ...;
PGPSecretKey secretKey = secretKeys.getSecretKey();
PGPPublicKey publicKey = secretKey.getPublicKey();
SecretKeyRingProtector protector = ...;

KeyRingInfo info = PGPainless.inspectKeyRing(secretKeys);

String userIdA = "A";
String userIdB = "B";
String userIdC = "C";
// Get current signatures for userIds
PGPSignature certificationA = info.getLatestUserIdCertification(userIdA);
PGPSignature certificationB = info.getLatestUserIdCertification(userIdB);
PGPSignature certificationC = info.getLatestUserIdCertification(userIdC);

PGPSignature newCertificationA = SignatureFactory.renewSelfCertification(secretKey, protector,
    new SelfSignatureSubpackets.Callback() {
        @Override
        modifyHashedSubpackets(SelfSignatureSubpackets subpackets) {
            subpackets.setPrimaryUserId(null); // override primary userid to make user-id non-primary
        }
    }, certificationA)
    .build(userIdA);

// insert signature into key
publicKey = PGPPublicKey.addCertification(publicKey, userIdA, newCertificationA);

PGPSignature newCertificationB = SignatureFactory.renewSelfCertification(secretKey, protector,
    new SelfSignatureSubpackets.Callback() {
        @Override
        modifyHashedSubpackets(SelfSignatureSubpackets subpackets) {
            subpackets.setPrimaryUserId(); // make user-id primary
        }
    }, certificationB)
    .build(userIdB);

// insert signature into key
publicKey = PGPPublicKey.addCertification(publicKey, userIdB, newCertificationB);

...
```

Of course, for common use cases much of this code can very easily be hidden behind convenience methods.

**Comment by IvanPizhenko** [comment]
Thank you. I think for now I will end up just with manual reordering of User Ids and setting the whole new list of User Ids.

**Comment by vanitasvitae** [comment]
Closed in favor of feature request #216
